### PR TITLE
fix: #1768 only allow select image file in ImageInput element

### PR DESCRIPTION
**Pull request by ant066**
# Pull request checklist

## Does this close any currently open issues?

fixes: #1768

**Please check if your PR fulfills the following requirements:**

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`yarn build`) was run locally and any changes were pushed
- [x] Lint (`yarn lint`) has passed locally and any fixes were made for failures
- [x] Test (`yarn test`) has passed locally and any fixes were made for failures

## Pull request type

**Please check the type of change your PR introduces:**

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Issue Number: #1768

## What is the new behavior?
Screen Shot 2020-06-24 at 11 54 28 AM

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/elements/src/utils/validators/validate-text-and-number.ts [hunk 1]
```diff
@@ -3,3 +3,8 @@ export const isTextAndNumberOnly = (value?: string): boolean => {
   const re = /^[a-zA-Z0-9|\s]+$/
   return re.test(value)
 }
+
+export const isImageType = (type: string) => {
```
Comment (vuhuucuong) [inline packages/elements/src/utils/validators/validate-text-and-number.ts:7]: Please move to `packages/utils` instead

File: packages/elements/src/utils/is-base64/index.ts [hunk 2]
```diff
@@ -8,3 +8,12 @@ export const isBase64 = (base64Str: string | undefined | null) => {
     return false
   }
 }
+
+export const getTypeFromBase64 = (base64: string): string => {
```
[the remark this answers is not in the document]
Comment (ant066) [reply packages/elements/src/utils/is-base64/index.ts]: just move it to @reapit/utils

File: packages/elements/src/utils/validators/validate-text-and-number.ts (hunk 1 above)
Comment (ant066) [reply packages/elements/src/utils/validators/validate-text-and-number.ts]: just move it to @reapit/utils

File: packages/elements/src/components/ImageInput/index.tsx [hunk 3]
```diff
@@ -1,15 +1,44 @@
...
+  const [message, setMessage] = useState<string>('')
+
+  const afterLoadedImage = (base64, handleClearFile) => {
+    if (!validateType(base64)) {
+      setMessage('You must select the image file!')
+      handleClearFile()
+      return
+    }
+    setMessage('')
+    props && props.afterLoadedImage && props.afterLoadedImage(base64)
```
Comment (vuhuucuong) [inline packages/elements/src/components/ImageInput/index.tsx:19]: props?.afterLoadedImage(base64) for short

File: packages/elements/src/components/ImageInput/index.tsx [hunk 4]
```diff
@@ -1,15 +1,44 @@
...
+      />
+      {message && <p className="has-text-danger">{message}</p>}
+    </>
+  )
+}
+
+export const validateType = (value: string): boolean => {
+  const fileType = getTypeFromBase64(value)
+  if (!isImageType(fileType)) return false
+  return true
```
Comment (vuhuucuong) [inline packages/elements/src/components/ImageInput/index.tsx:43]: Can we `return isImageType(fileType)` instead

**Pull request merged**